### PR TITLE
Get started: better dataset, dvclive, more plots

### PR DESCRIPTION
**Pull request by shcheklein**
Based on initial work done by @daavoo in this PR - https://github.com/iterative/example-repos-dev/pull/102 

Depends on https://github.com/iterative/dataset-registry/pull/28. A few things will be improved here when that PR lands.

Corresponding repo to play with: https://studio.iterative.ai/team/Iterative/views/example-get-started-zde16i6c4g

(a few lines in the PR will be uncommented / reverted before actual merge, after testing)

It changes example get started:

- To use a clean dataset (thus 2.5x smaller)
- Less number of features (3000 -> 100/200). Since dataset is better it's now enough - should be way better memory footprint (prepared matrix is 4x smaller)
- More plots - confusion matrix + forest analysis (most useful feature)
- CML report includes confusion matrix
- Two branches - more data, hyperparams tuning

### CODE REVIEW

File: example-get-started/generate.sh [hunk 1]
```diff
@@ -79,15 +79,16 @@ dvc push
 
 rm data/data.xml data/data.xml.dvc
 dvc import https://github.com/iterative/dataset-registry \
-  get-started/data.xml -o data/data.xml \
+  get-started/data.xml -o data/data.xml --rev 95d720c467496ea6c15dd2c5d5ad48bbb631d8b1 \
```
Comment (daavoo) [inline example-get-started/generate.sh:82]: Could we add some tags in the dataset-registry and use them as `rev`? 

Would be a great signal to have semantic meaning in the table (after https://github.com/iterative/studio/issues/2682 is implemented)

File: example-get-started/analyze.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import io
```
Comment (daavoo) [inline example-get-started/analyze.py:1]: Where / How is this file used?

File: example-get-started/code/.github/workflows/cml.yaml [hunk 3]
```diff
@@ -23,13 +23,17 @@ jobs:
...
           cml publish prc.svg --title "Precision & Recall" --md >> report.md
 
+          dvc plots show \
+            --show-vega evaluation/plots/predictions.json > vega.json
+          vl2svg vega.json confusion.svg
+          cml publish confusion.svg --title "Confusion Matrix" --md >> report.md
+
           echo "## Metrics" >> report.md
           echo "### $PREVIOUS_REF → workspace" >> report.md
           dvc metrics diff $PREVIOUS_REF --show-md >> report.md
```
Comment (daavoo) [inline example-get-started/code/.github/workflows/cml.yaml:37]:
```suggestion
          echo "## Metrics and Params" >> report.md
          echo "### $PREVIOUS_REF → workspace" >> report.md
          dvc exp diff $PREVIOUS_REF --show-md >> report.md
```

File: example-get-started/code/src/evaluate.py [hunk 4]
```diff
@@ -55,14 +58,20 @@
...
-        indent=4,
-    )
+
+# ... confusion matrix plot
+predictions = [{
+                "actual": int(actual),
+                "predicted": 1 if predicted > 0.5 else 0
+               } for actual, predicted in zip(labels, predictions)]
+with open("evaluation/plots/predictions.json", "w") as f:
+    json.dump(predictions, f)
```
Comment (daavoo) [inline example-get-started/code/src/evaluate.py:68]:
```suggestion
live.log_plot("confusion_matrix", labels, predictions_by_class.argmax(-1))
```

Comment (daavoo) [reply example-get-started/code/src/evaluate.py:68]: Would need to replace `predictions` with `confusion_matrix` elsewhere

File: example-get-started/code/src/requirements.txt [hunk 5]
```diff
@@ -1,4 +1,6 @@
+dvclive
```
Comment (daavoo) [inline example-get-started/code/src/requirements.txt:1]:
```suggestion
dvclive>=0.6
```

File: example-get-started/code/.github/workflows/cml.yaml [hunk 6]
```diff
@@ -23,13 +23,17 @@ jobs:
           echo "# CML Report" > report.md
           echo "## Plots" >> report.md
           dvc plots diff $PREVIOUS_REF workspace \
-            --show-vega --targets prc.json > vega.json
+            --show-vega --targets evaluation/plots/precision_recall.json > vega.json
           vl2svg vega.json prc.svg
           cml publish prc.svg --title "Precision & Recall" --md >> report.md
 
+          dvc plots show \
```
[the remark this answers is not in the document]
Comment (shcheklein) [reply example-get-started/code/.github/workflows/cml.yaml:30]: for a change primarily :) but also bc combined confusion matrix on diff doesn't look the same as a single one (when you merge they affect each other, colors will be different). I'm still trying to understand what is the best way to present them on diff. We keep them "merged" for now in Studio, but what is the best way? Not sure ...

File: example-get-started/generate.sh (hunk 1 above)
Comment (shcheklein) [reply example-get-started/generate.sh:82]: yes, this is until https://github.com/iterative/dataset-registry/pull/28, and there are few other places I'll revert in the PR

File: example-get-started/analyze.py (hunk 2 above)
Comment (shcheklein) [reply example-get-started/analyze.py:1]: @daavoo renamed the file, put a comment. This is a helper script, it's not part of the project. It used to generate a slice of data from the 100GB+ SO dump.

File: example-get-started/code/src/evaluate.py (hunk 4 above)
Comment (shcheklein) [reply example-get-started/code/src/evaluate.py:68]: done, thanks, I will give it a try

**Review by daavoo — APPROVED** [review]

File: example-get-started/code/src/evaluate.py [hunk 7]
```diff
@@ -55,14 +58,15 @@
...
-        },
-        fd,
-        indent=4,
-    )
+
+# ... confusion matrix plot
+live.log_plot("confusion_matrix", labels.squeeze(), predictions_by_class.argmax(-1))
+
+# ... and finally, we can dump an image, it's also supported:
+fig, axes = plt.subplots(dpi=100)
```
[the remark this answers is not in the document]
Comment (shcheklein) [reply example-get-started/code/src/evaluate.py:66]: @daavoo yep, I think we should also handle gracefully this on `dvc plots show` - images could be large after all, we should be normalizing their sizes

[the remark this answers is not in the document]
Comment (shcheklein) [reply example-get-started/code/src/evaluate.py:66]: and for VS Code it's better to have them bigger (you can click and zoom)

File: example-get-started/deploy.sh [hunk 8]
```diff
@@ -17,7 +17,9 @@ popd
 
 # Requires AWS CLI and write access to `s3://dvc-public/code/get-started/`.
 mv $PACKAGE_DIR/$PACKAGE .
-aws s3 cp --acl public-read $PACKAGE s3://dvc-public/code/get-started/$PACKAGE
+#aws s3 cp --acl public-read $PACKAGE s3://dvc-public/code/get-started/$PACKAGE
```
Comment (dberenbaum) [inline example-get-started/deploy.sh:20]: Is this intentionally left like this?

Comment (dberenbaum) [reply example-get-started/deploy.sh:20]: I get it for now, just making sure we don't forget to update it.

File: example-get-started/generate.sh [hunk 9]
```diff
@@ -79,15 +79,16 @@ dvc push
...
   --desc "Imported raw data (tracks source updates)"
 git add data/data.xml.dvc
 tick
 git commit -m "Import raw data (overwrite)"
 git tag -a "4-import-data" -m "Data file overwritten with an import."
 dvc push
 
-wget https://code.dvc.org/get-started/code.zip
+cp $HERE/code.zip .
+#wget https://code.dvc.org/get-started/code.zip
```
Comment (dberenbaum) [inline example-get-started/generate.sh:91]: Again, just making sure we don't forget to uncomment.

File: example-get-started/generate.sh [hunk 10]
```diff
@@ -212,9 +226,10 @@ gh repo create iterative/example-get-started --public \
 Run these commands to force push it:
 
 cd build/example-get-started
-git remote add origin git@github.com:iterative/example-get-started.git
+git remote add origin git@github.com:shcheklein/example-get-started.git
```
Comment (dberenbaum) [inline example-get-started/generate.sh:229]: Again, noting not to forget to update the remote url.

File: example-get-started/generate.sh [hunk 11]
```diff
@@ -127,27 +128,35 @@ git commit -m "Create ML pipeline stages"
...
-  --plots-no-cache roc.json \
-  python src/evaluate.py model.pkl data/features scores.json prc.json roc.json
-dvc plots modify prc.json -x recall -y precision
-dvc plots modify roc.json -x fpr -y tpr
-git add .gitignore dvc.yaml dvc.lock prc.json roc.json scores.json
+  -M evaluation.json \
+  --plots-no-cache evaluation/plots/precision_recall.json \
+  --plots-no-cache evaluation/plots/roc.json \
+  --plots-no-cache evaluation/plots/confusion_matrix.json \
+  --plots evaluation/importance.png \
```
Comment (dberenbaum) [inline example-get-started/generate.sh:138]: Not sure what's in scope, but should we add titles?

File: example-get-started/deploy.sh (hunk 8 above)
Comment (shcheklein) [reply example-get-started/deploy.sh:20]: yep, I'll remove it today before the merge

File: example-get-started/generate.sh (hunk 11 above)
Comment (shcheklein) [reply example-get-started/generate.sh:138]: specifically for this image? I had it, then decided that Y axis label is kinda enough to understand, wdyt?

Comment (dberenbaum) [reply example-get-started/generate.sh:138]: Yeah, I meant for the plots generally, but you are right. It took me a second to separate ROC/PRC, but a title will just duplicate the axis names.

File: example-get-started/generate.sh (hunk 10 above)
Comment (dberenbaum) [reply example-get-started/generate.sh:229]: Got it, makes sense!

**Pull request merged**